Wrappers/Python/cil/io/TXRMTXMDataReader.py

```diff
@@ -100,46 +128,59 @@ def read(self):
             DtoRADistance = dxchange.reader._read_ole_arr(ole, \
                     'ImageInfo/DtoRADistance', "<{0}f".format(number_of_images))
             
-        dist_source_center   = np.abs( StoRADistance[0] )
-        dist_center_detector = np.abs( DtoRADistance[0] )
-        
-        # normalise data by flatfield
-        data = data / metadata['reference']
-        
-        # circularly shift data by rounded x and y shifts
-        for k in range(number_of_images):
-            data[k,:,:] = np.roll(data[k,:,:], \
-                (int(metadata['x-shifts'][k]),int(metadata['y-shifts'][k])), \
-                axis=(1,0))
-        
-        # Pixelsize loaded in metadata is really the voxel size in um.
-        # We can compute the effective detector pixel size as the geometric
-        # magnification times the voxel size.
-        d_pixel_size = ((dist_source_center+dist_center_detector)/dist_source_center)*metadata['pixel_size']
-        
-        # convert angles to requested unit measure, Zeiss stores in radians
-        if self.angle_unit == AcquisitionGeometry.DEGREE:
-            angles = np.degrees(metadata['thetas'])
-        else:
-            angles = np.asarray(metadata['thetas'])
+            dist_source_center   = np.abs( StoRADistance[0] )
+            dist_center_detector = np.abs( DtoRADistance[0] )
 
-        self._ag = AcquisitionGeometry.create_Cone3D(
-            [0,-dist_source_center, 0] , [ 0, dist_center_detector, 0] \
-            ) \
-                .set_panel([metadata['image_width'], metadata['image_height']],\
-                    pixel_size=[d_pixel_size/1000,d_pixel_size/1000])\
-                .set_angles(angles, angle_unit=self.angle_unit)
-        self._ag.dimension_labels =  ['angle', 'vertical', 'horizontal']
+        if dist_source_center != dist_center_detector:
```

Comment:
Discussing with @gfardell it came out that it may be that ZEISS uses a virtual detector placed in the origin and scales the pixel sizes accordingly. Is this the case?

Reply:
Every parallel beam file I have checked, those distances both equal 1. In the cone beam files, the distances are never equal as that would likely mean the source and detector have collided. As for the virtual detector, I am not sure. Are we talking about data that is one geometry in particular or both?

Reply:
After looking at this some more, it seems for the parallel beam files the values are either both 1. in the TXRM file or 0. in TXM  file. I have not found any exception to this.

In the cone beam files, the value for `dist_source_center` in the TXRM file is in mm and has a negative sign in the and in the TXM file is in um and has a positive sign. For `dist_center_detector` the sign is the same in both but is in mm and um in the TXRM and TXM files respectively. 

I think since the values are always the same in the parallel beam files, this should be robust enough to parameterize this.

Reply:
Thanks. Could you please add a comment at the line explaining --as you've done here-- why we think the check is going to be sufficient and correct?

Reply:
Sorry for the long lead time on this. I thought on this some more and decided this was not enough. After looking at the metadata some more, I think we can use the `ImageInfo/AcquisitionMode` and `ImageInfo/XrayGeometry` streams. `ImageInfo/AcquisitionMode` is always 0 for txrm and 10 for txm files no matter the geometry. `ImageInfo/XrayGeometry`  either does not exist for parallel beam txrm files or is 0 for txm files. For both txrm and txm cone beam files this is always 1. This proved to be true for the newer files I have and the older files used in the demos so I think it should be robust enough. 

I also found the txrm2tiff package which was developed by researchers at the Diamond light source. Their metadata reader is much more comprehensive. I was thinking we could either request a change to their package and use it or even just ask if we can use the scripts/functions performing the reading and include them in CIL. What are your thoughts?